### ISSUE TITLE
Add apache pinot catalog, presto unable to start

### ISSUE THREAD

**Issue by sunny19930321** [prestodb/presto#13724]
I according to the document at the time of adding pinot connector, unable to start the presto (https://prestodb.io/docs/current/connector/pinot.html), because of less a configuration(pinot.controller-rest-service).
pinot.properties: 
connector.name=pinot
pinot.controller-urls=pinot-controller:9000
pinot.controller-rest-service=pinot-controller:9000

**Comment by shixuan-fan** [comment]
@agrawaldevesh (optional: @highker) Do you want to take a look?

**Comment by agrawaldevesh** [comment]
I will fix this.

**Comment by agrawaldevesh** [comment]
@sunny19930321 
> pinot.controller-rest-service=pinot-controller:9000

I think the official document https://prestodb.io/docs/current/connector/pinot.html does not make any reference to pinot.controller-rest-service. Please remove it from your pinot.properties file and retry. 

(This is needed only in Uber's internal environment and that's why we haven't widely publicized it in the document)

**Comment by sunny19930321** [comment]
> @sunny19930321
> 
> > pinot.controller-rest-service=pinot-controller:9000
> 
> I think the official document https://prestodb.io/docs/current/connector/pinot.html does not make any reference to pinot.controller-rest-service. Please remove it from your pinot.properties file and retry.
> 
> (This is needed only in Uber's internal environment and that's why we haven't widely publicized it in the document)

yes,  at first I did not configure pinot.controller-rest-service=pinot-controller:9000，launch presto server failed, and the log reported no  pinot.controller-rest-service. @agrawaldevesh 
![image](https://user-images.githubusercontent.com/20608540/69396695-1d4d3300-0d1e-11ea-8e84-8e81b089060b.png)

**Comment by sunny19930321** [comment]
I would like to ask presto connector pinot. When querying, will presto do a secondary merge after pinot aggregation querying?

**Comment by agrawaldevesh** [comment]
HI @sunny19930321 ... I am sorry about the issues you are facing. The attached PR would help. You can either apply it eagerly yourself or wait for it to be merged shortly. 

As for your question on the aggregation question: The aggregation is pushed into the pinot broker and in effect this becomes a single split query: the split doing the entire aggregation, that is pushed into the broker. The broker will then do the aggregation (pulling the data from the pinot servers and partially aggregating there as needed). 

does this answer your question about aggregation ? In pictures it would look like this:

presto-coordinator -> presto-worker -> pinot-broker -> N pinot-servers.

**Comment by sunny19930321** [comment]
> HI @sunny19930321 ... I am sorry about the issues you are facing. The attached PR would help. You can either apply it eagerly yourself or wait for it to be merged shortly.
> 
> As for your question on the aggregation question: The aggregation is pushed into the pinot broker and in effect this becomes a single split query: the split doing the entire aggregation, that is pushed into the broker. The broker will then do the aggregation (pulling the data from the pinot servers and partially aggregating there as needed).
> 
> does this answer your question about aggregation ? In pictures it would look like this:
> 
> presto-coordinator -> presto-worker -> pinot-broker -> N pinot-servers.

yes,  I found that the following aggregation query 1 was converted to no-aggregation query 2, 
the process is time-consuming.
query 1：
`SELECT 
    SUM(netflow) /1024 /1024 /1024 /300 AS netflow,
    fdate AS timeBucket
FROM       
    "cdn-all_cdn_staging" 
WHERE 
     fdate > 1573747200000
GROUP BY 
	 fdate
ORDER BY 
     timeBucket ASC`

`SELECT 
       fdate AS timeBucket, 
       SUM(CASE WHEN (hitinfo='Miss' or hitinfo LIKE '%Miss%') THEN request_num ELSE 0 END) * 1.000000 / SUM(request_num) AS miss_rate
FROM 
    "cdn-all_cdn_staging" 
WHERE 
     fdate > 1573747200000
GROUP BY 
	 fdate
ORDER BY 
     timeBucket ASC`

query 2:
`SELECT netflow, fdate FROM  cdn-all_cdn_staging WHERE  fdate > 1573747200000`

`SELECT request_num, hitinfo, fdate FROM cdn-all_cdn_staging_OFFLINE WHERE (fdate > 1573747200000)`

**Comment by agrawaldevesh** [comment]
Yeah I think you have hit the Achilles heel of the presto-pinot connector: 

In plain english: It will try to push down the computation into Pinot broker. But Pinot does not handle things like Case statements, multiple divisions etc. So the functions "SUM(netflow) /1024 /1024 /1024 /300" and "SUM(CASE WHEN hitinfo='Miss' THEN request_num ELSE 0 END) * 1.000000 / SUM(request_num)" are not being able to be pushed down into Pinot because pinot does not support multiple-divisions nor case statements.

So because the query cannot be pushed down, the Pinot connector is forced to pull the entire data into Presto (ie just do the basic filtering off the realtime/offline parts of the hybrid table) and then do the compute in presto. While this guarantees functionality, it makes the performance suffer.

**Comment by sunny19930321** [comment]
yes，thank you for your answer. I got it .@agrawaldevesh
